src/arch/ArchHooks/ArchHooks_Win32Static.cpp

```diff
@@ -9,38 +9,48 @@
 #include <cstdint>
 #include <vector>
 
-// for timeGetTime
+// for QueryPerformanceCounter
 #include <windows.h>
 #include <mmsystem.h>
 #if defined(_MSC_VER)
 #pragma comment(lib, "winmm.lib")
 #endif
 
-
 static bool g_bTimerInitialized;
 
+/* QueryPerformanceCounter variables below.
+ * QueryPerformanceCounter and QueryPerformanceFrequency expect
+ * a LARGE_INTEGER, which is a union. These functions store data
+ * in the QuadPart of the LARGE_INTEGER, which is type longlong. */
+LARGE_INTEGER g_bFrequency, g_bStartTime, g_bCurrentTime;
+double elapsedTime;
+
 static void InitTimer()
 {
 	if( g_bTimerInitialized )
 		return;
 	g_bTimerInitialized = true;
 
+	// Grab important information for QPC during the timer initialization
+	QueryPerformanceCounter(&g_bStartTime);
+	QueryPerformanceFrequency(&g_bFrequency);
+
+	// Set the system timer resolution to 1 millisecond
 	timeBeginPeriod( 1 );
```

Comment:
I don't think we need `timeBeginPeriod` at all so can likely remove if we're using QPC everywhere instead of `timeGetTime`. Might need to doublecheck though.

Reply:
Thanks! I thought so, but I wasn't sure, so I'll compile without it and be sure everything works the same to be sure.